leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisRegistrationStore.java

```diff
@@ -828,14 +828,22 @@ public void run() {
                         System.currentTimeMillis(), 0, cleanLimit);
 
                 for (byte[] endpoint : endpointsExpired) {
-                    Registration r = deserializeReg(j.get(toEndpointKey(endpoint)));
-                    if (!r.isAlive(gracePeriod)) {
-                        Deregistration dereg = removeRegistration(j, r.getId(), true);
-                        if (dereg != null)
-                            expirationListener.registrationExpired(dereg.getRegistration(), dereg.getObservations());
+                    byte[] regBytes = j.get(toEndpointKey(endpoint));
+                    if (regBytes == null) {
+                        LOG.warn("An expiration index points to an unknown registration: '{}'",
+                                new String(endpoint, UTF_8));
+                        j.zrem(EXP_EP, endpoint);
```

Comment:
:thinking:  a bit more about this but if this is a race condition like explained above :point_up:, should we really remove the expiration ?

Reply:
I wanted to remove the expiration because otherwise it will never be removed, no ?

Reply:
**IF** this is race condition between cleaner and `removeRegistration` **THEN** the zrem will be called just after. 

See 
```java 
j.del(toEndpointKey(r.getEndpoint()));  // ==> 1.  here we remove the registration
Collection<Observation> obsRemoved = unsafeRemoveAllObservations(j, r.getId());     
removeAddrIndex(j, r);
removeIdentityIndex(j, r);
removeExpiration(j, r);   /// ==> 2. here we call zrem
```
I guess cleaner is sometime called between this 2 steps  :thinking: 
That makes me think that `zrem` will be call anyway ?

Reply:
I start to think you are right about the race condition, because I don't see the error anymore and if the zrem was not done I would have this error in loop

Reply:
> if the zrem was not done I would have this error in loop

:thinking: that's make sense